### DISCUSSION TITLE
Use HIP1 for a new auction system

### DISCUSSION THREAD

**Discussion by Falci** [handshake-org/HIPs#37] [Brainstorming]
First, a quick overview of how HIP1 works:
1. Transfer the name to a lock script (prevents revoke)
2. Create a partial TX, signed by name owner, which specify which name is been transferred and the value
3. Owner needs to publish this TX somewhere (like shakedex.com)
4. A buyer can fund, sign the missing part of the TX and broadcast it. The swap is made in this step.

Here's my changes proposals:
  a. On step 1 we should add a time locker, so the step 4 can only be broadcasted after some time (fixed or relative)
  b. The name owner should be able to bypass this time locker
  b1. Alternatively, we could have another time locker, shorter, for the name owner
  c. After the TX is published (step 3) bidders can already bid. That will be temporary invalid, due the time locker, but we could also publish the "bid tx" in a public place (again, like shakedex.com)
  c1. This could be public or not
  d. Before the auction end (time lock) the owner can pick the winner TX. The incentive (but no guarantee) is the TX with higher value.

If the owner don't pick a winner, anyone can broadcast their TX after the time lock pass

Not sure if this is possible. WDYT?

**Comment by pinheadmz** [comment]
I think the problem with this is that nothing prevents "bidders" from broadcasting their tx as soon as locktime expires, basically how shakedex already works. The name owner has already signed the input with the name in it, they don't have the ability to sign anything else in the transaction, meaning their part of the deal is done. HIP6 covers this though because the name owner signs at the end of the protocol, after bidders have submitted bids.

**Reply by Falci** [reply]
The auction period is from the publish (step 3) to the time lock.
The winner needs to be declared (and tx broadcasted) before the time lock.

**Reply by pinheadmz** [reply]
Transactions are invalid before time lock though. They won't even relay.

**Reply by pinheadmz** [reply]
There is nothing the name owner can do to invalidate "losing" bids, and so you'll have a race condition.

**Reply by Falci** [reply]
> Transactions are invalid before time lock though. They won't even relay.

Check the item `b`

**Reply by Falci** [reply]
and if you are talking about bidders publishing their bids.. it should be done off chain

**Reply by pinheadmz** [reply]
> > Transactions are invalid before time lock though. They won't even relay.
> 
> Check the item `b`

Exactly. This is the thing you haven't fleshed out. In HIP1, the name owners signature is the first step of the protocol, they can't change anything after that. In HIP6, it's the last step of the protocol so they have an opportunity to manually pick a winner without any conflict.

**Reply by Falci** [reply]
What if we prepend the locking script is something like:
```
OP_IF
  <owner> OP_CHECKSIG
ELSE 
  <num> OP_ CSV
END_IF
```

**Reply by pinheadmz** [reply]
In HIP1, the locking script is unlocked by the presigns, so the owner will always be the signer. The whole point of HIP1 is that the name owner takes themself out of the equation immediately, they can not interact by design after they publish the presigns. By contrast the whole point of HIP6 is to put the owner back in control at the end of the auction ;-)
